### PR TITLE
more platform dependent conditional flags

### LINKED ISSUE #11: Two symbols missing in simulator setup
Hello,

SecKeychainGetTypeID and SecAccessGetTypeID are missing at link time when I target the iphone6 simulator. Strangely I don't have the problem when I target an actual iphone6. Do we have another cfg() mixup somewhere ?

```
undef: _SecKeychainGetTypeID
undef: _SecAccessGetTypeID
Undefined symbols for architecture x86_64:
  "_SecKeychainGetTypeID", referenced from:
      _$LT$security_framework..keychain..SecKeychain$u20$as$u20$core_foundation..base..TCFType$LT$$BP$mut$u20$security_framework_sys..base..OpaqueSecKeychainRef$GT$$GT$::type_id::h0b49430b489586d5 in SnipsSda(security_framework-d4c2fbca6aed6c3a.0.o)
  "_SecAccessGetTypeID", referenced from:
      _$LT$security_framework..access..SecAccess$u20$as$u20$core_foundation..base..TCFType$LT$$BP$mut$u20$security_framework_sys..base..OpaqueSecAccessRef$GT$$GT$::type_id::h5bfb1124c8dad652 in SnipsSda(security_framework-d4c2fbca6aed6c3a.0.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### PR DESCRIPTION
**Pull request by kali**
fixes #11

### CODE REVIEW

File: security-framework/src/lib.rs [hunk 1]
```diff
@@ -65,6 +67,7 @@ macro_rules! p {
     }
 }
 
+#[cfg(target_os = "macos")]
```
Comment (sfackler) [inline security-framework/src/lib.rs]: SecAccess does appear to be defined in the iOS SDK.

Comment (sfackler) [inline security-framework/src/lib.rs]: Oops, nevermind.

**Comment by sfackler** [comment]
Thanks!

@alexcrichton am I doing anything dumb with my iOS travis setup? I would have thought that it should have found this link error.

**Comment by alexcrichton** [comment]
@sfackler I'm actually not sure, the iOS stuff I did in liblibc was the first thing that worked, and I never bothered to figure out if it was actually testing what I wanted it to test.

**Pull request merged**